eng/source-build-patches/0017-Emit-RepoTasks.binlog.patch

```diff
@@ -0,0 +1,25 @@
+From 8f63217ad54ec431e398753b2bf745a1eb565e36 Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Mon, 21 Dec 2020 14:37:58 -0600
+Subject: [PATCH] Emit RepoTasks.binlog
+
+Lets us diagnose errors that occur while building RepoTasks.csproj.
+---
+ eng/build.sh | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/eng/build.sh b/eng/build.sh
+index 3c3cab39ce..029bb4f0c8 100755
+--- a/eng/build.sh
++++ b/eng/build.sh
+@@ -363,6 +363,7 @@ if [ "$build_repo_tasks" = true ]; then
+         -p:Restore=$run_restore \
+         -p:Build=true \
+         -clp:NoSummary \
++        -bl:RepoTasks.binlog \
+         ${toolset_build_args[@]+"${toolset_build_args[@]}"}
```

Comment:
/fyi we shouldn't take this approach. Instead
1. do a better job aligning how `-bl` is handled in eng/build.ps1 and eng/build.sh. The PowerShell version will add a unique binary log filename to this step as well as grab a log for the main build (if any).
2. make sure the file is placed in the usual artifacts/log/{configuration} folder.

Reply:
Agreed.